docs/lifecycle/hooks/hooks.md

```diff
@@ -0,0 +1,347 @@
+- Within the render funciton, we can still use the `this`context variable to access properties.
+
+## Directives and Data Binding
+
+This two concepts are essential when it comes to Vue.js. Let's first take a look at directives and then the way it's related to data binding:
+
+### Directives
+
+Similar to Angular.js/Angular, directives are special symbols we can find on HTML elements. This symbols or tokens tell Vue.js to do something special with the DOM element, for instance, set the innnerText, respond to an event and so on.
+
+In Vue.js, the convention is to use `v-`as the prefix for every directive. Let's take a look to a very simple example:
+
+
+```html
+<template> 
+  <div v-text="greeting"></div>
+</template>
+```
+
+```javascript
+module.exports = {
+  data: function() {
+    return {
+      greeting: "Hello Gorilla Logic!"
+    };
+  },
+}
+```
+
+In the example above, we're telling Vue to render the text of the ``` div ```element using a directive ``` v-text ``` which is tied to the data property ```greeting ```.
+
+```html
+<template> 
+  <div v-on:click="clickHandler"></div>
+</template>
+```
+
+```javascript
+module.exports = {
+  methods: {
+    clickHandler(){
+      console.log('Clicked using a directive');
+    },
+  },
+}
+```
+
+In the example above, we're telling Vue to execute a method called ```clickHandler ``` when user clicks the div. The syntax ``` v-on:click ``` help us to achieve this.
+
+### Data Binding
+
+As you migh have noticed, directives are useful to achieve multiple things such as configure event handlers, determine element behavior and bind data. Data binding is a very specific and useful way to render data in components, even more because it takes advantage of Vue.js reactivity implementation to react to data changes. 
+
+In the first exampple of data biding, we can expect the innerText of the div DOM element to refresh every time the ``` greeting ```property changes.
+
+## Event handling
+
+As previously mentioned during the directives explanation, there are special directives in Vue.js that allow us to listen events in DOM elements and eventually execute actions using the power of javascript. Javascript code that reacts to the event can be expressed in multiple ways, for example using inline expressions like ```v-on:clicl="counter += 1" ``` or even component methdos like this ``` v-on:click="methodA" ```, where ``` methodA ``` is a defined component's method.
+
+
+## Event propagation management
+
+When working with events, it's important to have control to event behaviors. For instance, when we need to stop the default propagation of a submit event using ```e.preventDefault() ```. Even though we can achieve this using the default approach inside our events, we take advantage of **Event Modifiers** provided by Vue.js. In simple words, event modifiers are suffixes that we can place on even directives in order to determine the behavior of the event a its propagation like this:
```

Comment:
Didn't catch this. Specially on the highlighted text. Probably some rewording is needed.

_In simple words, event modifiers are suffixes that we can place **on even directives** in order to determine the behavior **of the event a its propagation** like this:_

Reply:
You're right, the sentence is confusing. I restated all the setence like this:

"_When working with events, it's important to have control over event behaviors. For instance, when we need to stop the default propagation of a submit event, it's pretty common to proceed with a  ```e.preventDefault() ``` expression as the first line of our event handler. Although this approach is fine, we can also take advantage of the **Event Modifiers**. This are nothing but suffixes we place right after the event directives, that allow us to control the propagation of the event itself (without even placing additional logic into our event modifier), for instance_"